dbms/src/Core/SettingsCommon.cpp

```diff
@@ -63,6 +63,17 @@ void SettingNumber<Type>::set(const String & x)
     set(parse<Type>(x));
 }
 
+template <>
+void SettingNumber<bool>::set(const String & x)
+{
+    if (x == "false")
+        set(false);
+    else if (x == "true")
+        set(true);
+    else
+        set(parse<bool>(x));
```

Comment:
It seems it's better to support case-insensitive comparison add  checks for "YES" and "NO" and throw exceptions on unexpected input.